test/apps/test_fuzzy_match.py

```diff
@@ -1,36 +1,51 @@
 #!/usr/bin/python2
 
+#
+# Copyright (C) 2021 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# See file LICENSE for terms.
+
 import os
 import commands
 import re
 import argparse
 import sys
 import logging
 
-###################
-# Main test runner
-###################
-class FuzzyTester:
-    def __init__(self, ucx_info, verbose):
-        self.ucx_info = ucx_info
-        if verbose:
-            logging.basicConfig(level=logging.DEBUG)
+class FuzzyTest(object):
+    '''Handles test attributes (environment variables, expected results)'''
+    def __init__(self, expected):
+        logging.info('Using env vars: %s' % expected.keys())        
+        self.expected = expected
         
-    def cleanup(self):
+    def __enter__(self):
+        for var_name in self.expected:
```

Comment:
The `__enter__` and `__exit__` are "asymmetrical":
you add vars from `self.expected`, but remove vars starting with `UCX_`. These 2 sets can differ.

Reply:
that because you also want to clean UCX vars defined by the parent process